### PR TITLE
docs: document release procedures

### PR DESCRIPTION
**Pull request by mattnichols**
# Summary of Changes

Documenting release and versioning strategy.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,8 +32,48 @@ dependencies {
...
+
+#### feature branches
+
+  1. Developers create feature branches to introduce code changes
+  2. When complete, a Pull Request is created from the branch
+  3. Once approved, the pull request is merged into `master`
+
+#### release branches
+
+When work on a _planned_ major version is ready to start, a release branch will be created to hold the in-progress the work. Multiple feature branches and pull requests will be generated against this branch until it the version is ready for release.
```
Comment (stevecl5) [inline README.md:54]:
```suggestion
When work on a _planned_ major version is ready to start, a release branch will be created to hold the in-progress work. Multiple feature branches and pull requests will be generated against this branch until the version is ready for release.
```
Grammar

File: README.md [hunk 2]
```diff
@@ -32,8 +32,48 @@ dependencies {
...
+We use semantic versioning. git tags are placed at the point (SHA) where the version was generated.
+
+_Current Major Version_: On the _current major version_, minor and patch versions can be generated at any time and are expected to be backward compatible.
+
+_Past Major Versions_: If a severe issue is discovered in a past major version, a release branch will be created off of the last patch version and a service pack will be released (example: 2.3.2-sp.1) from that branch.
+
+_Next Major Version_: Major version releases can come follow 2 different paths:
+
+  1. Planned (typical): When a large upgrade is planned, a release branch will be generated to hold the new code away from master as it is built.
+  2. Unplanned: Some breaking changes may a). be deemed necessary for the health of the project or b). be of minor impact to users of the library. These changes will follow the normal flow and will be merged directly to _master_ (from feature branches).
```
Comment (stevecl5) [inline README.md:67]:
```suggestion
  2. Unplanned: Some breaking changes may a) be deemed necessary for the health of the project or b) be of minor impact to users of the library. These changes will follow the normal flow and will be merged directly to _master_ (from feature branches).
```
Suggestion. I think the extra periods look weird.

File: README.md [hunk 3]
```diff
@@ -32,8 +32,48 @@ dependencies {
...
+
+Regardless of the type, when a major version is ready to be published, a release candidate will be generated (example: 3.0.0-rc.1). This is considered an optional, production-ready release but should be used with some caution. Once the release candidate has been vetted in production, a new release will be generated without the release candidate designation (in the case of a planned version, the version branch will be merged to master at this time).
+
+
 ## Contributing
-Create a topic branch, make your changes, and create an PR.
+
+Commits must conform to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification. The commit types are used to automate version bumps, using [release-please](https://github.com/googleapis/release-please).
+
+To contribute a change, create a feature branch off or master or the current release branch. Commit changes to branch. Push branch up and create a pull request.
```
Comment (stevecl5) [inline README.md:76]:
```suggestion
To contribute a change, create a feature branch off of master or the current release branch. Commit changes to branch. Push branch up and create a pull request.
```
Typo

**Review by stevecl5 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (mattnichols) [reply README.md:54]: I don't know how many times I've read this. Still missed those: 🤦

Comment (mattnichols) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (mattnichols) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (mattnichols) [reply README.md]: Fixed.

**Review by stevecl5 — APPROVED** [review]

**Pull request merged**